### PR TITLE
astyle space-before-comma to chill the buildbot

### PR DESCRIPTION
**Pull request by keyspace**
Also one occurence of brace-on-wrong-line.

All touched files in `astyle_whitelist`.

These are about half of the diffs produced by `make astyle`.

### CODE REVIEW

**Comment by keyspace** [comment]
One of those build glitches that "fixed by itself"?..

File: src/profession.cpp [hunk 1]
```diff
@@ -221,7 +221,7 @@ void profession::check_item_definitions( const itypedecvec &items ) const
 {
     for( auto &itd : items ) {
         if( !item::type_is_defined( itd.type_id ) ) {
-            debugmsg( "profession %s: item %s does not exist", id.c_str() , itd.type_id.c_str() );
+            debugmsg( "profession %s: item %s does not exist", id.c_str(), itd.type_id.c_str() );
```
Comment (sethsimon) [inline src/profession.cpp]: Strange, the origin of the extra space is the ancient 8888c70. What version of astyle does the build bot use? Maybe it was upgraded recently?

Also, there seems to be an astyle regression in profession.h (see the build log of #20206) (so why did this build not fail?!).

File: src/morale.cpp [hunk 2]
```diff
@@ -480,7 +480,8 @@ void player_morale::display( double focus_gain )
             const int decimals = ( value - static_cast<int>( value ) != 0.0 ) ? 2 : 0;
             color = ( value > 0.0 ) ? c_green : c_red;
             mvwprintz( w, y, getmaxx( w ) - 8, color, "%+6.*f", decimals, value );
-        } else {
+        } else
+        {
```
Comment (Coolthulhu) [inline src/morale.cpp]: This is from astyle? Since when is it supposed to put opening brace on the new line?

Comment (keyspace) [inline src/morale.cpp]: Tried again - yes. `astyle` version 2.06. :/

File: src/profession.cpp (hunk 1 above)
Comment (keyspace) [inline src/profession.cpp]: I have no idea.

By the time I submitted this, PRs were no longer failing en-masse.

The situation here is very similar to #20093.

File: src/morale.cpp (hunk 2 above)
Comment (keyspace) [inline src/morale.cpp]: Tried moving the curly to its proper place after the `if` a few lines above - onto the second line. `make astyle` moves it right back where it is now. :/

I now think I should close this PR, until someone merges it by accident.

**Comment by narc0tiq** [comment]
> One of those build glitches that "fixed by itself"?..

Not so much; I was messing around with the build script to make it work on both the old and new Jenkinses, and one of the things I needed was to keep the `ASTYLE_BINARY` setting in the Jenkins config rather than the [`mainline-build.zsh`](http://dev.narc.ro/cataclysm/mainline-build.zsh).

Sadly, I forgot to update the PR builder config to match, so it was trying to use astyle 2.04 for validation (and failing silently because "unable to check" == "everything's fine"). Soon, that will be getting taken into account.

---

Here's a fun problem -- `make astyle` gives me this diff:

```diff
diff --git i/src/profession.h w/src/profession.h
index a71c1ead35..5067414225 100644
--- i/src/profession.h
+++ w/src/profession.h
@@ -19,10 +19,10 @@ class JsonArray;
 class JsonObject;
 class addiction;
 enum add_type : int;
-class Skill;
-using skill_id = string_id<Skill>;
+    class Skill;
+    using skill_id = string_id<Skill>;

-class profession
+    class profession
 {
     public:
         typedef std::pair<skill_id, int> StartingSkill;
```

Why would `astyle` want to add a bunch of leading whitespace there? Is it not pleased with the enum's forward declaration? This is the only thing that's failing the builds right now.

**Comment by narc0tiq** [comment]
Oh, huh: https://github.com/CleverRaven/Cataclysm-DDA/pull/19924/files#r99763073

I mean, now I know what to fix, at least.

**Comment by keyspace** [comment]
You're correct, the situation's very different from #20093.

What `astyle` is it using now? `2.06` is [relatively new](http://astyle.sourceforge.net/notes.html) - from Dec 2016. `2.05.1` was Dec 2014, `2.04` was Nov 2013.

File: src/morale.cpp (hunk 2 above)
Comment (keyspace) [inline src/morale.cpp]: Just checked: this is the only place in the entire codebase where:

1. the file is in `astyle_whitelist`;
2. the block is a lambda function with a return value;
3. the function body has an `else` clause.

There is a similar place in `src/game_inventory.cpp`, around line 380, that has (1) and (2) but not (3). Giving it an `else` so it does have (3) does not result in the above behaviour.

So there must be other conditions than the three above, like being nested in a specific block...

The very fact that

``` c++
         if( value != 0.0 )
         {
```

is OK in `astyle`'s book raises a flag.

**Comment by kevingranade** [comment]
All we can do for these for the time being is remove these files from the
astyle whitelist and wait for them to fix the bug.

**Pull request closed without merging**